pacopy/natural.py

```diff
@@ -42,8 +43,11 @@ def natural(
             to bootstrap the Newton process for the next iteration (order 0). Another
             possibility is to use :math:`u - s J^{-1}(u, \\lambda)
             \\frac{df}{d\\lambda}`, a first-order approximation.
+        milestones (Optional[Iterable[float]]): Don't step over these values.
     """
     lmbda = lambda0
+    if milestones is not None:
```

Comment:
```
milestones = [] if milestones is None else milestones
```

Reply:
Ah.  The motivation for wrapping `milestones` in `iter` is that a user will probably pass either a `list` or a `numpy.ndarray`, as in

https://github.com/nschloe/pacopy/blob/e2d972fcb80c3c43efdeb30e89f78bd83cfd1ad9/test/test_bratu1d.py#L83

and one can't call `next` on either of those.  

Lists and NumPy ndarrarys are `Iterable` 
```python
isinstance(np.array([8e2]), typing.Iterable)
```
but not *iterators*.

Reply:
There's a catch here in dispensing with the new name `lambdas` for the processed argument; see [below](https://github.com/nschloe/pacopy/pull/6#issuecomment-481933802).

Reply:
Note that `iter` is idempotent and therefore harmless in the case where a user does actually pass an iterator.

A reasonable use-case there is to generate an infinite sequence of milestones, e.g. passing `milestones=itertools.count()`, presumably terminating with `max_steps` #8 or by raising an `Exception` in the `callback`.